Seems like this PR breaks some examples. Can you fix them?

```
+ go build ./_example/advanced_bot ./_example/buttons ./_example/commands ./_example/sharded ./_example/simple ./_example/undeleter
# github.com/diamondburned/arikawa/v3/_example/buttons
_example/buttons/main.go:38:6: cannot use option.NewNullableString("This is a message with a button!") (type *option.NullableStringData) as type string in field value
_example/buttons/main.go:39:6: cannot use &[]discord.Component{...} (type *[]discord.Component) as type []discord.Component in field value
_example/buttons/main.go:88:5: cannot use option.NewNullableString("Custom ID: " + e.Data.CustomID) (type *option.NullableStringData) as type string in field value
# github.com/diamondburned/arikawa/v3/_example/commands
_example/commands/main.go:36:5: cannot use option.NewNullableString("Pong!") (type *option.NullableStringData) as type string in field value
```

(see CI logs)

--- a/_example/buttons/main.go
+++ b/_example/buttons/main.go
@@ -9,7 +9,6 @@ import (
 	"github.com/diamondburned/arikawa/v3/discord"
 	"github.com/diamondburned/arikawa/v3/gateway"
 	"github.com/diamondburned/arikawa/v3/session"
-	"github.com/diamondburned/arikawa/v3/utils/json/option"
 )
 
 // To run, do `APP_ID="APP ID" GUILD_ID="GUILD ID" BOT_TOKEN="TOKEN HERE" go run .`
@@ -35,8 +34,8 @@ func main() {
 			data := api.InteractionResponse{
 				Type: api.MessageInteractionWithSource,
 				Data: &api.InteractionResponseData{
-					Content: option.NewNullableString("This is a message with a button!"),
-					Components: &[]discord.Component{
+					Content: "This is a message with a button!",
+					Components: []discord.Component{
 						discord.ActionRowComponent{
 							Components: []discord.Component{
 								discord.ButtonComponent{
@@ -85,7 +84,7 @@ func main() {
 		data := api.InteractionResponse{
 			Type: api.UpdateMessage,
 			Data: &api.InteractionResponseData{
-				Content: option.NewNullableString("Custom ID: " + e.Data.CustomID),
+				Content: "Custom ID: " + e.Data.CustomID,
 			},
 		}
 

--- a/_example/commands/main.go
+++ b/_example/commands/main.go
@@ -9,7 +9,6 @@ import (
 	"github.com/diamondburned/arikawa/v3/discord"
 	"github.com/diamondburned/arikawa/v3/gateway"
 	"github.com/diamondburned/arikawa/v3/session"
-	"github.com/diamondburned/arikawa/v3/utils/json/option"
 )
 
 // To run, do `APP_ID="APP ID" GUILD_ID="GUILD ID" BOT_TOKEN="TOKEN HERE" go run .`
@@ -33,7 +32,7 @@ func main() {
 		data := api.InteractionResponse{
 			Type: api.MessageInteractionWithSource,
 			Data: &api.InteractionResponseData{
-				Content: option.NewNullableString("Pong!"),
+				Content: "Pong!",
 			},
 		}
 

--- a/api/interaction.go
+++ b/api/interaction.go
@@ -55,18 +55,15 @@ type InteractionResponseData struct {
 	//
 	// Required: one of content, file, embeds
 	Content string `json:"content,omitempty"`
-
 	// TTS is true if this is a TTS message.
 	TTS bool `json:"tts,omitempty"`
 	// Embeds contains embedded rich content.
 	//
 	// Required: one of content, file, embeds
 	Embeds []discord.Embed `json:"embeds,omitempty"`
-
 	// Components is the list of components (such as buttons) to be attached to
 	// the message.
 	Components []discord.Component `json:"components,omitempty"`
-
 	// AllowedMentions are the allowed mentions for the message.
 	AllowedMentions *AllowedMentions `json:"allowed_mentions,omitempty"`
 	// Flags are the interaction application command callback data flags.
@@ -116,8 +113,8 @@ func (c *Client) RespondInteraction(
 	return sendpart.POST(c.Client, resp, nil, URL)
 }
 
-// OriginalInteractionResponse returns the initial interaction response.
-func (c *Client) OriginalInteractionResponse(
+// InteractionResponse returns the initial interaction response.
+func (c *Client) InteractionResponse(
 	appID discord.AppID, token string) (*discord.Message, error) {
 
 	var m *discord.Message
@@ -152,8 +149,8 @@ func (data EditInteractionResponseData) WriteMultipart(body *multipart.Writer) e
 	return sendpart.Write(body, data, data.Files)
 }
 
-// EditOriginalInteractionResponse edits the initial Interaction response.
-func (c *Client) EditOriginalInteractionResponse(
+// EditInteractionResponse edits the initial Interaction response.
+func (c *Client) EditInteractionResponse(
 	appID discord.AppID,
 	token string, data EditInteractionResponseData) (*discord.Message, error) {
 
@@ -181,8 +178,8 @@ func (c *Client) EditOriginalInteractionResponse(
 		EndpointWebhooks+appID.String()+"/"+token+"/messages/@original")
 }
 
-// DeleteOriginalInteractionResponse deletes the initial interaction response.
-func (c *Client) DeleteOriginalInteractionResponse(appID discord.AppID, token string) error {
+// DeleteInteractionResponse deletes the initial interaction response.
+func (c *Client) DeleteInteractionResponse(appID discord.AppID, token string) error {
 	return c.FastRequest("DELETE",
 		EndpointWebhooks+appID.String()+"/"+token+"/messages/@original")
 }

--- a/api/interaction.go
+++ b/api/interaction.go
@@ -67,13 +67,14 @@ type InteractionResponseData struct {
 	// the message.
 	Components []discord.Component `json:"components,omitempty"`
 
-	// Files represents a list of files to upload. This will not be
-	// JSON-encoded and will only be available through WriteMultipart.
-	Files []sendpart.File          `json:"-"`
-	Flags InteractionResponseFlags `json:"flags,omitempty"`
-
 	// AllowedMentions are the allowed mentions for the message.
 	AllowedMentions *AllowedMentions `json:"allowed_mentions,omitempty"`
+	// Flags are the interaction application command callback data flags.
+	Flags InteractionResponseFlags `json:"flags,omitempty"`
+
+	// Files represents a list of files to upload. This will not be
+	// JSON-encoded and will only be available through WriteMultipart.
+	Files []sendpart.File `json:"-"`
 }
 
 // NeedsMultipart returns true if the InteractionResponseData has files.
@@ -89,6 +90,27 @@ func (d InteractionResponseData) WriteMultipart(body *multipart.Writer) error {
 // an "interaction callback".
 func (c *Client) RespondInteraction(
 	id discord.InteractionID, token string, resp InteractionResponse) error {
+
+	if resp.Data.Content == "" && len(resp.Data.Embeds) == 0 && len(resp.Data.Files) == 0 {
+		return ErrEmptyMessage
+	}
+
+	if resp.Data.AllowedMentions != nil {
+		if err := resp.Data.AllowedMentions.Verify(); err != nil {
+			return errors.Wrap(err, "allowedMentions error")
+		}
+	}
+
+	sum := 0
+	for i, embed := range resp.Data.Embeds {
+		if err := embed.Validate(); err != nil {
+			return errors.Wrap(err, "embed error at "+strconv.Itoa(i))
+		}
+		sum += embed.Length()
+		if sum > 6000 {
+			return &discord.OverboundError{Count: sum, Max: 6000, Thing: "sum of all text in embeds"}
+		}
+	}
 
 	var URL = EndpointInteractions + id.String() + "/" + token + "/callback"
 	return sendpart.POST(c.Client, resp, nil, URL)
@@ -116,6 +138,8 @@ type EditInteractionResponseData struct {
 	// Attachments are the attached files to keep
 	Attachments *[]discord.Attachment `json:"attachments,omitempty"`
 
+	// Files represents a list of files to upload. This will not be
+	// JSON-encoded and will only be available through WriteMultipart.
 	Files []sendpart.File `json:"-"`
 }
 
@@ -131,8 +155,7 @@ func (data EditInteractionResponseData) WriteMultipart(body *multipart.Writer) e
 // EditOriginalInteractionResponse edits the initial Interaction response.
 func (c *Client) EditOriginalInteractionResponse(
 	appID discord.AppID,
-	token string, data EditInteractionResponseData,
-) (*discord.Message, error) {
+	token string, data EditInteractionResponseData) (*discord.Message, error) {
 
 	if data.AllowedMentions != nil {
 		if err := data.AllowedMentions.Verify(); err != nil {
@@ -185,7 +208,7 @@ func (c *Client) CreateInteractionFollowup(
 		}
 		sum += embed.Length()
 		if sum > 6000 {
-			return nil, &discord.OverboundError{sum, 6000, "sum of all text in embeds"}
+			return nil, &discord.OverboundError{Count: sum, Max: 6000, Thing: "sum of all text in embeds"}
 		}
 	}
 
@@ -225,6 +248,7 @@ func (c *Client) EditInteractionFollowup(
 // DeleteInteractionFollowup deletes a followup message for an interaction
 func (c *Client) DeleteInteractionFollowup(
 	appID discord.AppID, messageID discord.MessageID, token string) error {
+
 	return c.FastRequest("DELETE",
 		EndpointWebhooks+appID.String()+"/"+token+"/messages/"+messageID.String())
 }

